kratos/tests/test_skin_detection_process.py

```diff
@@ -122,10 +96,74 @@ def test_NotOnSubModelPartSkinDetectionProcess(self):
         }"""))
         detect_skin.Execute()
 
-        ## DEBUG
-        #self._post_process(model_part)
+        # # DEBUG
+        # self._post_process(self.model_part)
+
+        # # DEBUG
+        # self._post_process_mpi(self.model_part)
 
-        self.assertEqual(model_part.NumberOfConditions(), 112)
+        # Check the number of conditions created
+        data_comm = self.model_part.GetCommunicator().GetDataCommunicator()
+        if data_comm.IsDistributed():
+            number_of_conditions = 0
+            number_of_conditions += self.model_part.NumberOfConditions()
+            global_number_of_conditions = data_comm.SumAll(number_of_conditions)
+            self.assertEqual(global_number_of_conditions, 112)
+        else:
+            self.assertEqual(self.model_part.NumberOfConditions(), 112)
```

Comment:
```suggestion
        # Check the number of conditions created
        self.assertEqual(model_part.GetCommunicator().GlobalNumberOfConditions(), 112)
```

Reply:
Doesn't work, I don't know why, maybe I need to use DistributedModelPartInitializer first, not very practical